### PR TITLE
Add recipe for uSockets C/C++ library

### PR DESCRIPTION
**Pull request by traversaro**
[uSockets](https://github.com/uNetworking/uSockets) is a C/C++ library for sockets, a dependency of the [uWebSockets](https://github.com/uNetworking/uWebSockets) library for dealing with websockets in C++.

Repology page: https://github.com/uNetworking/uSockets .

As by default uSockets ships with a minimal Makefile that does not support shared builds, this PR provides a minimal CMake build system to support all the different configuration required by conda-forge, as done by vcpkg: https://github.com/microsoft/vcpkg/tree/2022.11.14/ports/usockets .

The library requires the use of macOS SDK 10.13, so for now the macOS build is disabled and it will be re-enabled once the feedstock is created. 

The recipe name contains the `lib` prefix as requested in https://github.com/microsoft/vcpkg/tree/2022.11.14/ports/usockets for C/C++ libraries.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by traversaro** [comment]
@conda-forge/help-c-cpp The recipe is ready for review.

File: recipes/libusockets/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# See https://github.com/conda-forge/nodejs-feedstock/blob/6e33145837f3d719fed66d020e848d1636e55e38/recipe/conda_build_config.yaml#L19
+pin_run_as_build:
+  # libuv is missing run_exports
+  libuv:
+    max_pin: x.x
+    min_pin: x.x.x
```
Comment (carterbox) [inline recipes/libusockets/conda_build_config.yaml:6]: What's the difference between pinning in conda_build_config and using the pin_compatible jinja expression?

**Review by carterbox — APPROVED** [review]

File: recipes/libusockets/conda_build_config.yaml (hunk 1 above)
Comment (traversaro) [reply recipes/libusockets/conda_build_config.yaml:6]: To be honest, I am not sure. I took this approach from the linked nodejs-feedstock, but I did not considered alternative approaches.

**Comment by traversaro** [comment]
Thanks @carterbox !

**Pull request merged**